code/modules/clothing/modular_armor/attachments/modules.dm

```diff
@@ -475,6 +475,86 @@
 	shield_color_mid = LIGHT_COLOR_RED_ORANGE
 	shield_color_full = LIGHT_COLOR_ELECTRIC_CYAN
 
+/obj/item/armor_module/module/mirage
+	name = "\improper Loki Illusion Module"
+	desc = "Designed for mounting on modular armor. This module creates a holographic projection of the user, which can be used to distract enemies and draw their fire when the user is hit."
+	icon = 'icons/mob/modular/modular_armor_modules.dmi'
+	icon_state = "mod_illusion"
+	worn_icon_state = "mod_illusion_a"
+	slowdown = 0
+	attach_features_flags = ATTACH_REMOVABLE|ATTACH_ACTIVATION|ATTACH_APPLY_ON_MOB
+	slot = ATTACHMENT_SLOT_MODULE
+	toggle_signal = COMSIG_KB_ARMORMODULE
+	COOLDOWN_DECLARE(mirage_cooldown)
+
+/obj/item/armor_module/module/mirage/activate(mob/living/user)
+	if(!COOLDOWN_CHECK(src, mirage_cooldown))
+		balloon_alert(user, "[COOLDOWN_TIMELEFT(src, mirage_cooldown)/10] seconds")
+		return
+	var/alpha_mod = user.alpha * 0.8
+	user.alpha -= alpha_mod
+	var/mob/illusion/mirage_nade/fake = new(get_turf(user), user, null, 15 SECONDS)
+	addtimer(CALLBACK(src, PROC_REF(end_mirage), user, alpha_mod, fake), 15)
+	COOLDOWN_START(src, mirage_cooldown, 30 SECONDS)
+
+/// just cleans up the alpha on both the user and the fake
+/obj/item/armor_module/module/mirage/proc/end_mirage(mob/user, alpha_mod, mob/illusion/mirage_nade/fake)
+	user.alpha += alpha_mod
```

Comment:
this will fuckout with other things that hard set alpha, like scout cloak.

Reply:
Honestly a problem with scout cloak, it should be doing like I do here with multing it or using a proper api instead of hard set, it will fuck with other sources because of that rn anyway